### ISSUE TITLE
Chapter 2 - Same data used for training and predictions

### ISSUE THREAD

**Issue by urvishmhjn** [ageron/handson-ml2#254]
```python
lin_reg = LinearRegression()
lin_reg.fit(housing_prepared, housing_labels)
housing_predictions = lin_reg.predict(housing_prepared)
```

The `housing_prepared` data is used for both training and then prediction purposes. This has been done in all the ML models. 
Shouldn't this practice give misleading/wrong results? 
This should also give us wrong values of mean squared error.

Please clarify the doubt.
_(Refer Chapter 2 - Pages 75, 76)_

**Comment by Praful932** [comment]
@urvishmhjn Predictions are just used for calculating the training accuracy to see how well the model is doing on the training data.

**Comment by Aliiiu** [comment]
pls my lin_reg.fit is giving svd related error

**Comment by ageron** [comment]
Hi @urvishmhjn ,
Thanks for your question. As @Praful932 said, it is useful to look at the performance of the model on various datasets, including the training set. If the performance is low on the training set, it means that the model is underfitting. Once the model is no longer underfitting, you can compare the score on the training set with the score on the validation set (or test set). The `cross_val_score()` used a bit further in the book uses k-fold cross-validation to evaluate the model in a fair way (never evaluating the model on data it was trained on).
Hope this helps
